core/src/main/java/org/elasticsearch/index/shard/ShardStateMetaData.java

```diff
@@ -100,6 +110,9 @@ public void toXContent(XContentBuilder builder, ShardStateMetaData shardStateMet
             builder.field(VERSION_KEY, shardStateMetaData.version);
             builder.field(PRIMARY_KEY, shardStateMetaData.primary);
             builder.field(INDEX_UUID_KEY, shardStateMetaData.indexUUID);
+            if (shardStateMetaData.allocationId != null) {
```

Comment:
I would prefer to have a comment instead (on the read method), so it's clear it can be removed in future versions